mlir/lib/Analysis/Presburger/Barvinok.cpp

```diff
@@ -245,3 +246,235 @@ QuasiPolynomial mlir::presburger::detail::getCoefficientInRationalFunction(
   }
   return coefficients[power].simplify();
 }
+
+/// Substitute x_i = t^μ_i in one term of a generating function, returning
+/// a quasipolynomial which represents the exponent of the numerator
+/// of the result, and a vector which represents the exponents of the
+/// denominator of the result.
```

Comment:
what is the semantics of the vector for the denominator? is the denominator a sum of the coefficients?